Zend/zend_language_scanner.l

```diff
@@ -1422,7 +1422,7 @@ NEWLINE ("\r"|"\n"|"\r\n")
 	RETURN_TOKEN_WITH_IDENT(T_RETURN);
 }
 
-<ST_IN_SCRIPTING>"@@" {
+<ST_IN_SCRIPTING>"@[" {
```

Comment:
This is missing `enter_nesting('[');` to account for the closing `]` of an attribute.

Reply:
Ok, really? The original #[ didn't have that either and patch works this way.

Reply:
Look [here](https://github.com/koolkode/php-src/commit/b4f63d54b2f5e15b15ec9a9f207315b075531cd1), in this case `yytext[1]` is `[`. I am pretty much sure that this was the first thing I had to add when I started to implement `#[`.

Reply:
its fascinating, because it compiled and tests ran for me without this...

Reply:
The nesting checks were [added](https://github.com/php/php-src/commit/80598f12507c5cbde04163289e4d2575f05d2a0c) to PHP in April. Maybe you have used an older branch / master for testing?

Reply:
Does not fully work either Parse error: syntax error, unexpected token "function", expecting "]" in /home/travis/build/php/php-src/Zend/tests/attributes/005_objects.php on line 26